nupic/research/frameworks/sigopt/mixins/imagenet_params.py

```diff
@@ -0,0 +1,131 @@
+        # field called max_epochs.
+        if suggestion.task is not None and "epoch" in suggestion.task.name:
+            max_epochs = self.sigopt_config["metadata"]["max_epochs"]
+            epochs = int(max_epochs * suggestion.task.cost)
+            print("Suggested task/cost/epochs for this multitask experiment: ",
+                  suggestion.task.name, suggestion.task.cost, epochs)
+            config["epochs"] = epochs
+            config["lr_scheduler_args"]["epochs"] = epochs
+
+        # Optimizer args
+        if "log_lr" in assignments:
+            config["optimizer_args"]["lr"] = math.exp(assignments["log_lr"])
+            assignments.pop("log_lr")
+
+        if "momentum" in assignments:
+            config["optimizer_args"]["momentum"] = assignments["momentum"]
+            config["lr_scheduler_args"]["max_momentum"] = assignments["momentum"]
+            assignments.pop("momentum")
+
+        if "weight_decay" in assignments:
+            config["optimizer_args"]["weight_decay"] = assignments["weight_decay"]
+            assignments.pop("weight_decay")
+
+        # lr_scheduler args
+        if "gamma" in assignments:
+            config["lr_scheduler_args"]["gamma"] = assignments["gamma"]
+            assignments.pop("gamma")
+
+        if "step_size" in assignments:
+            config["lr_scheduler_args"]["step_size"] = assignments["step_size"]
+            assignments.pop("step_size")
+
+        # Parameters for OneCycleLR
+        if "pct_start" in assignments:
+            # Ensure integer number of epochs in pct start to avoid crash.
+            start_epochs = int(assignments["pct_start"] * config["epochs"] + 0.5)
+            pct_start = float(start_epochs) / config["epochs"]
+            config["lr_scheduler_args"]["pct_start"] = pct_start
+            assignments.pop("pct_start")
+
+        if "cycle_momentum" in assignments:
+            config["lr_scheduler_args"]["cycle_momentum"] = \
+                eval(assignments["cycle_momentum"])
+            assignments.pop("cycle_momentum")
+
+        if "max_lr" in assignments or "init_lr" in assignments:
+            # From the OneCycleLR docs:
+            #   initial_lr = max_lr/div_factor
+            #   min_lr = initial_lr/final_div_factor
+            max_lr = assignments.get("max_lr", 6.0)
+            init_lr = assignments.get("init_lr", 1.0)
+
+            config["lr_scheduler_args"]["max_lr"] = max_lr
+            config["lr_scheduler_args"]["div_factor"] = max_lr / init_lr
+            config["lr_scheduler_args"]["final_div_factor"] = init_lr / 0.00025
+            assignments.pop("init_lr", None)
+            assignments.pop("max_lr", None)
+
+        config.update(assignments)
+
+    @classmethod
+    def get_execution_order(cls):
+        eo = super().get_execution_order()
+        name = "ImagenetParams"
+        eo["update_config_with_suggestion"] = [name + ".update_config_with_suggestion"]
```

Comment:
Why not `append`?

Reply:
I have it overwrite it for full control. I expect it to be used only for experiments within our imagenet project. I may make this more explicit. Possibly, I can move this mixin to the projects folder to imply it is custom made for those experiments and shouldn't be generally applied.